test/parser/test_type_currency.py

```diff
@@ -11,31 +11,35 @@ class _TestModel(BaseModel):
 @pytest.mark.parametrize(
     "input_value, expected_output",
     [
-        ("$1,000.00", 1000.00),
-        ("€1.00", 1.0),
-        ("£1,000,000.00", 1000000.00),
-        ("¥1000.00", 1000.00),
-        ("₹1000", 1000.00),
-        (1000, 1000.00),
-        (1000.00, 1000.00),
-        ("-¥1,234.56", -1234.56),
-        ("$0.1", 0.1),
-        ("€.1", 0.1),
-        ("1000", 1000.00),
-        ("$1,234.5678", 1234.5678),
-        ("   $1,234  ", 1234.0),
-        ("   $ 1,234  ", 1234.0),
-        ("1.234 €", 1234.0),
-        ("1.234€", 1234.0),
-        ("€1.234", 1234.0),
-        ("1,234$", 1234.0),
-        ("1,234 $", 1234.0),
-        ("1.234,45", 1234.45),
-        ("1.234.456,00", 1234456.00),
-        ("1.000.000", 1000000.00),
-        ("Starting At 12.99", 12.99),
-        ("From 399.99", 399.99),
-        ("0.0004 $", 0.0004),
+        ("$1,000.00", "$1000.0"),
+        ("€1.00", "€1.0"),
+        ("£1,000,000.00", "£1000000.0"),
+        ("¥1000.00", "¥1000.0"),
+        ("₹1000", "₹1000.0"),
+        (1000, "1000.0"),
+        (1000.00, "1000.0"),
+        (123.45, "123.45"),
+        ("-¥1,234.56", "-¥1234.56"),
+        ("-$123.12", "-$123.12"),
+        ("-€123.123,12", "-€123123.12"),
+        ("$0.1", "$0.1"),
+        ("€.1", "€0.1"),
+        ("1000", "1000.0"),
+        ("$1,234.5678", "$1234.5678"),
+        ("   $1,234  ", "$1234.0"),
+        ("   $ 1,234  ", "$1234.0"),
+        ("1.234 €", "€1234.0"),
+        ("1.234€", "€1234.0"),
+        ("€1.234", "€1234.0"),
+        ("1,234$", "$1234.0"),
+        ("1,234 $", "$1234.0"),
+        ("1.234,45", "1234.45"),
+        ("1.234.456,00", "1234456.0"),
+        ("1.000.000", "1000000.0"),
+        ("Starting At 12.99", "12.99"),
+        ("From 399.99", "399.99"),
+        ("From 399.99$", "$399.99"),
+        ("0.0004 $", "$0.0004"),
```

Comment:
Personally i think currecy should be outputed as an array:
{
   "currency": "USD"
   "price": 12.99
}

Reply:
![Screenshot 2024-10-02 at 3 47 55 PM](https://github.com/user-attachments/assets/74600ef1-8c60-41b5-ac0d-539573fe505f)
 example from zyte

Reply:
hmmm yeah, asim and i discussed this and we will create a new ticket for this once agreed on the best object representation